### PR TITLE
Adopt Swift 6 and audit for Sendable

### PR DESCRIPTION
**Pull request by parkera**
This patch:

* Adopts Swift 6 mode for the libraries, executables, and tests
* Updates all types for `Sendable` annotation, and adds a flag to require all future types to do so as well
* Syncs with Darwin Foundation for `noasync` functions
* Addresses all Sendable warnings. 
  * In many cases this is done by using `nonisolated(unsafe)`. This should be considered a short term workaround and not a design pattern. In the future, adding these will be disfavored and rightfully proving to the compiler that it is safe will be preferred. However, it's not practical to do all of that in one shot.
  * Many warnings were also addressed by use of new `Synchronization` types and more annotations.
* Addresses all remaining warnings of all other types.

### CODE REVIEW

**Comment by parkera** [comment]
I'm not sure how reviewable this is. It is the result of repeated iteration and small fixes.

**Review by jmschonfeld — COMMENTED** [review]
I just did a quick skim since as you mentioned it's a large patch that's not easily reviewable, but I did catch one thing that might be worth double checking

File: Sources/Foundation/AffineTransform.swift [hunk 1]
```diff
@@ -412,7 +412,7 @@ public struct NSAffineTransformStruct {
     }
 }
 
-open class NSAffineTransform: NSObject {
+open class NSAffineTransform: NSObject, @unchecked Sendable {
```
[the remark this answers is not in the document]
Comment (parkera) [reply Sources/Foundation/AffineTransform.swift]: Done

**Comment by parkera** [comment]
@swift-ci please test

**Comment by parkera** [comment]
@swift-ci test

**Comment by Azoy** [comment]
https://github.com/apple/swift-foundation/pull/742
@swift-ci please test

**Comment by jmschonfeld** [comment]
@Azoy the linked PR only fixed the Synchronization import for `FoundationEssentials`/`FoundationInternationalization`. We'll need to make the same changes to `Foundation`/`FoundationNetworking`/`FoundationXML` here before importing `Synchronization` from these modules will work

**Comment by parkera** [comment]
@swift-ci test

**Comment by parkera** [comment]
@swift-ci test

**Comment by parkera** [comment]
@swift-ci test

**Comment by parkera** [comment]
@swift-ci test

**Comment by parkera** [comment]
@swift-ci test windows

**Pull request merged**